src/inet/InetInterface.cpp

```diff
@@ -186,6 +186,25 @@ bool InterfaceAddressIterator::HasBroadcastAddress()
     return HasCurrent() && (otIp6GetMulticastAddresses(Inet::globalOtInstance) != nullptr);
 }
 
+CHIP_ERROR InterfaceIterator::GetInterfaceType(InterfaceType & type)
+{
+    type = InterfaceType::Thread;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR InterfaceIterator::GetHardwareAddress(uint8_t * addressBuffer, uint8_t & addressSize, uint8_t addressBufferSize)
+{
+    VerifyOrReturnError(addressBuffer != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(addressBufferSize >= sizeof(otExtAddress), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    const otExtAddress * extendedAddr = otLinkGetExtendedAddress(Inet::globalOtInstance);
```

Comment:
Minor: Probably you could use ThreadStackMgr().GetPrimary802154MACAddress instead of raw OT api, as we already have method to get this address.